lib/fs.js

```diff
@@ -2070,11 +2094,16 @@ function writeAll(fd, isUserFd, buffer, offset, length, signal, callback) {
     } else {
       offset += written;
       length -= written;
-      writeAll(fd, isUserFd, buffer, offset, length, signal, callback);
+      writeAll(
+        fd, isUserFd, buffer, offset, length, signal, encoding, callback);
     }
   });
 }
 
+function isCustomIterable(obj) {
+  return isIterable(obj) && !isArrayBufferView(obj) && typeof obj !== 'string';
+}
+
```

Comment:
Should we extract `isCustomIterable` into `internal/fs/utils.js`? I think it was used in

https://github.com/nodejs/node/blob/18e4f405b14b26f971d5f108b92224e7c1fe34c9/lib/internal/fs/promises.js#L733

Reply:
Thanks. I fixed it.